pandas/io/parquet.py

```diff
@@ -250,14 +252,37 @@ def read(
             if dtype_backend == "pandas":
                 result = pa_table.to_pandas(**to_pandas_kwargs)
             elif dtype_backend == "pyarrow":
-                result = DataFrame(
-                    {
-                        col_name: arrays.ArrowExtensionArray(pa_col)
-                        for col_name, pa_col in zip(
-                            pa_table.column_names, pa_table.itercolumns()
-                        )
-                    }
-                )
+                index_columns = pa_table.schema.pandas_metadata.get("index_columns", [])
+                result_dc = {
+                    col_name: arrays.ArrowExtensionArray(pa_col)
```

Comment:
Is there a reason to not use `types_mapper` here (as is done for our own masked nullable arrays)? 

Because all this handling of the index is done by pyarrow already, and if using `to_pandas()` as for the other code paths would ensure this is done consistently.

Reply:
This is a good idea, but it looks like arrow is not applying the types mapper to index either. Not sure why we didn't do it like this initially though.

I tried:

```
types_mapper = lambda x: pd.ArrowDtype(x)
```

Reply:
I switched to ``types_mapper=pd.ArrowDtype`` in #51766

If you have converting the Index on your agenda, I'd avoid implementing this ourselves for a couple of weeks basically.

Reply:
From my limited understanding of https://github.com/apache/arrow/blob/main/python/pyarrow/src/arrow/python/arrow_to_pandas.cc I used the manual conversion to avoid a conversion from numpy(?) 

> // Functions for pandas conversion via NumPy

Reply:
They are going through ``pandas_dtype.__from_arrow__(arr)`` which receives an arrow array, so we should be good?

Reply:
Yes, whenever pyarrow detects an extension dtype for a column (either in the metadata, the pyarrow type itself or types_mapper keyword), we don't actually convert to numpy but directly pass the pyarrow array to `dtype.__from_arrow__`

Reply:
Got it thanks for the confirmation. I think https://github.com/pandas-dev/pandas/pull/51766 should be sufficient then